core/src/impl/Kokkos_Combined_Reducer.hpp

```diff
@@ -266,13 +265,26 @@ struct CombinedReducerImpl<std::integer_sequence<size_t, Idxs...>, Space,
     return m_value_view;
   }
 
-  KOKKOS_FUNCTION
+  template <typename ExecutionSpace, int Idx, class View>
```

Comment:
Pick one, `typename` or `class`, but do not mix
```suggestion
  template <typename ExecutionSpace, int Idx, class View>
```